WordPress/src/test/java/org/wordpress/android/ui/jetpack/backup/download/usecases/GetBackupDownloadStatusUseCaseTest.kt

```diff
@@ -0,0 +1,128 @@
+import org.wordpress.android.ui.jetpack.backup.download.BackupDownloadRequestState.Progress
+import java.util.Date
+
+@InternalCoroutinesApi
+class GetBackupDownloadStatusUseCaseTest : BaseUnitTest() {
+    private lateinit var useCase: GetBackupDownloadStatusUseCase
+    @Mock lateinit var networkUtilsWrapper: NetworkUtilsWrapper
+    @Mock lateinit var activityLogStore: ActivityLogStore
+    @Mock private lateinit var site: SiteModel
+    @Mock lateinit var statusModel: BackupDownloadStatusModel
+
+    @Before
+    fun setup() {
+        useCase = GetBackupDownloadStatusUseCase(networkUtilsWrapper, activityLogStore)
+        whenever(networkUtilsWrapper.isNetworkAvailable()).thenReturn(true)
+        whenever(activityLogStore.getBackupDownloadStatusForSite(site)).thenReturn(statusModel)
+    }
+
+    @Test
+    fun `given site, when status is fetched, then NetworkUnavailable is returned on no network`() = test {
+        whenever(networkUtilsWrapper.isNetworkAvailable()).thenReturn(false)
+
+        val result = useCase.getBackupDownloadStatus(site, downloadId).toList(mutableListOf())
+
+        Assertions.assertThat(result).contains(Failure.NetworkUnavailable)
+    }
+
+    @Test
+    fun `given site, when status is fetched, then RemoteRequestFailure is returned on failure`() = test {
+        whenever(activityLogStore.fetchBackupDownloadState(any())).thenReturn(
+                OnBackupDownloadStatusFetched(BackupDownloadStatusError(GENERIC_ERROR), FETCH_BACKUP_DOWNLOAD_STATE)
+        )
+
+        val result = useCase.getBackupDownloadStatus(site, downloadId).toList(mutableListOf())
+
+        Assertions.assertThat(result).contains(RemoteRequestFailure)
+    }
+
+    @Test
+    fun `given site, when status is fetched, then Complete is returned on success`() = testWithSuccessResponse {
+        whenever(activityLogStore.getBackupDownloadStatusForSite(site)).thenReturn(completeStateModel)
+        val result = useCase.getBackupDownloadStatus(site, downloadId).toList(mutableListOf())
+
+        Assertions.assertThat(result).contains(completeStatus)
+    }
+
+    @Test
+    fun `given site, when status is fetched, then Empty is returned when empty`() = testWithSuccessResponse {
+        whenever(activityLogStore.getBackupDownloadStatusForSite(site)).thenReturn(null)
+        val result = useCase.getBackupDownloadStatus(site, downloadId).toList(mutableListOf())
+
+        Assertions.assertThat(result).contains(Empty)
+    }
+
+    @Test
+    fun `given progress is not null, when backup is running, then Progress is returned`() = testWithSuccessResponse {
+        whenever(activityLogStore.getBackupDownloadStatusForSite(site))
+                .thenReturn(inProgressModel)
+                .thenReturn(completeStateModel)
+        val result = useCase.getBackupDownloadStatus(site, downloadId).toList(mutableListOf())
+
+        Assertions.assertThat(result).contains(progressStatus)
```

Comment:
Question (❓): I am guessing this test is kind of incomplete, right? Would the `completeStatus` need to be asserted as well, since it is used in the `thenReturn(...)`? Like this: `assertThat(result).contains(progressStatus, completeStatus)`

Reply:
That's there to get us out of the progress loop, but it won't hurt. 👍

Reply:
Ah, I see now, thanks for sharing! 🌟